### PR TITLE
Add guard for when ``name`` is ``None``

### PR DESCRIPTION
**Pull request by jay3332**
Pray that I got it correct because I need to sleep

### CODE REVIEW

File: ferrischat_webserver/src/guilds/edit_guild.rs [hunk 1]
```diff
@@ -13,32 +13,44 @@ pub async fn edit_guild(
...
+    sqlx::query!(
+        "SELECT * FROM guilds WHERE id = $1",
         bigint_guild_id
     )
     .fetch_optional(db)
     .await;
+    
+    match name {
+        Some(name) => {
+            let db = get_db_or_fail!();
```
Comment (tazz4843) [inline ferrischat_webserver/src/guilds/edit_guild.rs:26]: Move this back to the root of the function that way you only fetch it once.

File: ferrischat_webserver/src/guilds/edit_guild.rs [hunk 2]
```diff
@@ -13,32 +13,44 @@ pub async fn edit_guild(
...
-    let resp = sqlx::query!(
-        "UPDATE guilds SET name = $1 WHERE id = $2 RETURNING *",
-        name,
+    
+    sqlx::query!(
+        "SELECT * FROM guilds WHERE id = $1",
         bigint_guild_id
     )
     .fetch_optional(db)
     .await;
```
Comment (tazz4843) [inline ferrischat_webserver/src/guilds/edit_guild.rs:22]: There's no point in fetching this until all of the updates are done.

File: ferrischat_webserver/src/guilds/edit_guild.rs [hunk 3]
```diff
@@ -13,32 +13,44 @@ pub async fn edit_guild(
...
+    
+    match name {
+        Some(name) => {
+            let db = get_db_or_fail!();
+
+            let resp = sqlx::query!(
+                "UPDATE guilds SET name = $1 WHERE id = $2 RETURNING *",
```
Comment (tazz4843) [inline ferrischat_webserver/src/guilds/edit_guild.rs:29]: Don't have it return anything. Like I said on Discord, do a separate query to fetch the object after everything is done.

File: ferrischat_webserver/src/guilds/edit_guild.rs [hunk 4]
```diff
@@ -13,32 +13,44 @@ pub async fn edit_guild(
...
+                },
+                Err(e) => HttpResponse::InternalServerError().json(InternalServerErrorJson {
+                    reason: format!("DB returned an error: {}", e),
+                }),
+            }   
         },
-        Err(e) => HttpResponse::InternalServerError().json(InternalServerErrorJson {
-            reason: format!("DB returned an error: {}", e),
-        }),
+        None => HttpResponse::BadRequest().finish()
```
Comment (tazz4843) [inline ferrischat_webserver/src/guilds/edit_guild.rs:54]: This shouldn't be a 400, the field is Option for a reason.

File: ferrischat_webserver/src/guilds/edit_guild.rs [hunk 5]
```diff
@@ -13,32 +13,44 @@ pub async fn edit_guild(
...
-                message: "Guild not found".to_string(),
-            }),
+            match resp {
+                Ok(resp) => match resp {
+                    Some(guild) => HttpResponse::Ok().json(Guild {
+                        id: bigdecimal_to_u128!(guild.id),
+                        owner_id: bigdecimal_to_u128!(guild.owner_id),
+                        name: guild.name.clone(),
+                        channels: None,
+                        members: None,
```
Comment (tazz4843) [inline ferrischat_webserver/src/guilds/edit_guild.rs:43]: Do another query to fetch the updated guild object after you've done the query to update it.

**Review by tazz4843 — CHANGES_REQUESTED** [review]
For examples of what I mean, check out [`ferrischat_webserver/src/users/edit_user.rs`](https://github.com/FerrisChat/Server/blob/develop/ferrischat_webserver/src/users/edit_user.rs)

**Comment by Drxckzyz** [comment]
wow this needs alot of Work

**Comment by hydrostaticcog** [comment]
Stale

**Pull request closed without merging**